icu4c/source/i18n/units_router.cpp

```diff
@@ -43,8 +43,24 @@ Precision UnitsRouter::parseSkeletonToPrecision(icu::UnicodeString precisionSkel
     return result;
 }
 
+UnitsRouter::UnitsRouter(const char *inputUnitIdentifer, StringPiece region, StringPiece usage,
+                         UErrorCode &status) {
+    auto inputUnit = MeasureUnit::forIdentifier(inputUnitIdentifer, status);
+    this->init(inputUnit, region, usage, status);
+}
+
 UnitsRouter::UnitsRouter(MeasureUnit inputUnit, StringPiece region, StringPiece usage,
                          UErrorCode &status) {
+    this->init(inputUnit, region, usage, status);
+}
+
+void UnitsRouter::init(MeasureUnit inputUnit, StringPiece region, StringPiece usage,
```

Comment:
I believe this will result in an unnecessary copy of a MeasureUnit. Make inputUnit a const reference?

Reply:
The old signature takes a MeasureUnit by value, too.  We just need to make sure we `move` into this function rather than copy.

Reply:
copying behaviour is intended because the users of the `UnitsRouter` could continue the input object of `MeasureUnit`.

Reply:
or we can replace it with `const MeasureUnit&`

Reply:
This was what I was suggesting: making it a const reference. init doesn't keep a copy of inputUnit, it only uses it to produce a MeasureUnitImpl, and that via "forMeasureUnitMaybeCopy", which means it's not referencing inputUnit's fImpl. A const reference is then most efficient, to avoid ever making a copy of the MeasureUnit.

Reply:
Changing the signure from by-value to a const reference shouldn't impact client code.

(Marking thread as unresolved to help make these two new comments of mine more noticeable.)

Reply:
done!